frontend/src/types/semester.ts

```diff
@@ -0,0 +1,12 @@
+export interface Semester {
```

Comment:
according to kenneth chen these shouldn't be exported

Reply:
according to kenneth chen these files should also be named with the extension .d.ts

Reply:
Literally couldn't use my interface without exporting it, so I'm not going to remove it for now